mongo_connector/dest_mapping.py

```diff
@@ -15,143 +15,223 @@
 import logging
 import threading
 import re
+
+from collections import namedtuple, MutableSet
+
 from mongo_connector import errors
 
 
 LOG = logging.getLogger(__name__)
 
-"""New structure to handle both plain and
-wildcard mapping namespaces dynamically.
-"""
 
+_MappedNamespace = namedtuple('MappedNamespace',
+                              ['name', 'include_fields', 'exclude_fields'])
 
-class DestMapping():
-    def __init__(self, namespace_set, ex_namespace_set, user_mapping):
-        """
-            namespace_set and ex_namespace_set will not be non-empty
-            in the same time.
-            user_mapping should be non-empty together with namespace_set.
-            This assumption has been verified in the get_config_options()
-            in connector.py.
-        """
+
+class MappedNamespace(_MappedNamespace):
+    def __new__(cls, name=None, include_fields=None, exclude_fields=None):
+        return super(MappedNamespace, cls).__new__(
+            cls, name, include_fields, exclude_fields)
```

Comment:
include_fields/exclude_fields will be used in the next PR to add filtering fields per namespace.